### PR TITLE
Zachmu/merge upstream

### PR DESCRIPTION
**Pull request by zachmu**
Merging from upstream

### CODE REVIEW

**Comment by zachmu** [comment]
This is a merge from upstream/master, as well as a few changes to the parser to get rid of shift / reduce errors. Mainly pay attention to sql.y and parse_test.go.

Not sure why the generated sql.go doesn't match the one in source. Maybe because it's a merge instead of an overwrite? I'll fix it on master after this merge if it LGTY.

File: go/vt/sqlparser/sql.y [hunk 1]
```diff
@@ -461,6 +468,10 @@ delete_statement:
   {
     $$ = &Delete{Comments: Comments($2), Targets: $3, TableExprs: $5, Where: NewWhere(WhereStr, $6)}
   }
+|DELETE comment_opt delete_table_list from_or_using table_references where_expression_opt
```
Comment (reltuk) [inline go/vt/sqlparser/sql.y]: `| DELETE`...

File: go/vt/sqlparser/sql.y [hunk 2]
```diff
@@ -2178,6 +2206,12 @@ table_name:
     $$ = TableName{Qualifier: $1, Name: $3}
   }
 
+delete_table_name:
+table_id '.' '*'
+  {
+    $$ = TableName{Name: $1}
```
Comment (reltuk) [inline go/vt/sqlparser/sql.y]: Is this not a database name here?

Comment (reltuk) [reply go/vt/sqlparser/sql.y]: I wrong...

File: go/vt/sqlparser/sql.y [hunk 3]
```diff
@@ -486,6 +497,16 @@ table_name_list:
     $$ = append($$, $3)
   }
 
+delete_table_list:
```
Comment (reltuk) [inline go/vt/sqlparser/sql.y]: Should this just include the `table_name` case as well, with the `DELETE comment_opt table_name_list` case above being dropped? It seems like this fails to parse `DELETE some_stuff.*, some_other_stuff FROM some_stuff, some_other_stuff;`

**Pull request merged**